Sources/PianoRoll/PreferenceKeys.swift

```diff
@@ -0,0 +1,15 @@
+import SwiftUI
+
+/// For accumulating note view offsets.
+public struct NoteOffsetsKey: PreferenceKey {
+    public static var defaultValue: [NoteOffsetInfo] = []
+
+    public static func reduce(value: inout [NoteOffsetInfo], nextValue: () -> [NoteOffsetInfo]) {
+        value.append(contentsOf: nextValue())
```

Comment:
should reduce append to a list or replace a value?

Reply:
Since we want to have an array of note offsets so that we can scroll to any note with its offset, we should append to a list.

This is similar to what `Keyboard` has done https://github.com/AudioKit/Keyboard/blob/fb198a5355d7e276e63e891ec5805f2b2ab0780d/Sources/Keyboard/PreferenceKeys.swift
https://github.com/AudioKit/Keyboard/blob/fb198a5355d7e276e63e891ec5805f2b2ab0780d/Sources/Keyboard/KeyContainer.swift#L37-L41